src/sentry/migrations/0429_backfill_substatus_for_groupstatus_ignored.py

```diff
@@ -0,0 +1,106 @@
+# Generated by Django 2.2.28 on 2023-04-25 22:40
+
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.models import GroupStatus, GroupSubStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapper
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_groupedmessage
+    SET substatus = %s
+    FROM (VALUES %s) as data (id)
+    WHERE sentry_groupedmessage.id = data.id
+"""
+
+
+def backfill_substatus(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+    GroupSnooze = apps.get_model("sentry", "GroupSnooze")
+
+    cursor = connection.cursor()
+    archived_forever_batch = []
+    archived_until_condition_met_batch = []
+
+    for group_id, status, substatus in RangeQuerySetWrapper(
+        Group.objects.all().values_list("id", "status", "substatus"),
+        result_value_getter=lambda item: item[0],
+    ):
+        if status is not GroupStatus.IGNORED:
+            continue
+
+        group_snooze = GroupSnooze.objects.filter(group_id=group_id)
+
+        if not group_snooze:
+            archived_forever_batch.append((group_id, GroupSubStatus.FOREVER))
+        if substatus is not None:
+            archived_until_condition_met_batch.append(
+                (group_id, GroupSubStatus.UNTIL_CONDITION_MET)
+            )
```

Comment:
Should this be an `if .. else`? Otherwise you're setting different substatus values for the same group.